iceoryx_binding_c/source/cpp2c_subscriber.cpp

```diff
@@ -14,17 +14,17 @@
 
 #include "iceoryx_binding_c/internal/cpp2c_subscriber.hpp"
 
-bool cpp2c_Subscriber::setConditionVariable(iox::popo::ConditionVariableData* const conditionVariableDataPtr) noexcept
+void cpp2c_Subscriber::setConditionVariable(iox::popo::ConditionVariableData* const conditionVariableDataPtr) noexcept
```

Comment:
This can not fail. If the condition variable is already set then the already set condition variable is overridden and the condition is detached from the previous waitset.